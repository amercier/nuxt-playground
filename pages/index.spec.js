import Vue from 'vue';
import index from './index.vue';

function getRenderedText(Component, propsData) {
  const Ctor = Vue.extend(Component);
  const vm = new Ctor({ propsData }).$mount();
  return vm.$el.textContent;
}

describe('Index Component', () => {
  let $mounted;

  beforeEach(() => {
    $mounted = new Vue(index).$mount();
  });

  test('snapshot', () => {
    const $html = $mounted.$el.outerHTML;
    expect($html).toMatchSnapshot();
  });

  test('renders correctly with different props', () => {
    expect(getRenderedText(index, {})).toBe('Hello world!');
  });

  afterEach(() => {
    $mounted = null;
  });
});
