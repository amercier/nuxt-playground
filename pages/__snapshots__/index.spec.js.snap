// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Component snapshot 1`] = `"<h1 class=\\"grey\\">Hello world!</h1>"`;
